full spelling and grammar check/and correction (#71)
MillenniumMachines/millenniummachines.github.io: docs/millennium-os/manual/chapters/40_gcodes.md

Reset the stored details for a given WCS, or the current WCS if not specified.

> Different fields are used for different types of probing operations, and we want to reset these values before running a probing cycle, so if previous values existed, but the probing cycle failed we would not end up using the previously valid values.

Reviewer: I'm not sure about this change to `previously` - the values are still valid, they're just from an 'old' probe cycle. `previously valid` sounds like the values were valid but are now not.

Maybe there's another way to write this that is clearer, without using previous or previously?

Author: The whole thing could use a bit of a rewrite for clarity. See if the below is a correct understanding of what's meant to be there:

> Not all fields are needed for every probing operation type, and we want to reset the old values before running a new probing cycle. If previous values exist and the probing cycle fails, we will not use the existing values.

Reviewer: ```suggestion
Not all fields are needed for every probing operation type, and we want to avoid using previously-stored values if the most recent probing operation has failed. To do that, we use this macro to clear the relevant values before starting each probing cycle.
```

Does this work? I think you're right about needing a rewrite but I'm struggling with the "existing values" thing.

M5011 W2
```

> Look up stored rotation values from the given workplace number or the current workplace number if none is specified. If a value exists, the operator will prompted to apply rotation compensation using the inbuilt `G68` command.

Reviewer: ```suggestion
Look up stored rotation values from the given workplace number or the current workplace number if none is specified. If a value exists, the operator will be prompted to apply rotation compensation using the inbuilt `G68` command.
```

```gcode
M8004 Knn

; Wait for general purpose input 0 to change the state
M8004 K0
```

> Waits for a general-purpose input pin to change state. The state that is changed is unimportant, just that the state changes. This code is used by `M3.9` and `M5.9` when spindle feedback is enabled to wait until the VFD reports that it has reached the target speed.

Reviewer: ```suggestion
Waits for a general-purpose input pin to change state. The level of the pin is unimportant, just that the state changes. This code is used by `M3.9` and `M5.9` when spindle feedback is enabled to wait until the VFD reports that it has reached the target speed.
```

M7
```

> Enables the general-purpose output associated with unpressurised coolant, set by the operator during the configuration wizard. If air blast (`M7.1`) is not already activated, it will be before activating the coolant output.

Reviewer: ```suggestion
Enables the general-purpose output associated with unpressurised coolant, set by the operator during the configuration wizard. Air blast (`M7.1`) will be activated before the coolant output, if necessary.
```

; Stop all coolants
M9

; Restore all coolants to their previously saved states
M9 R1
```

> By default, this turns off any enabled coolant outputs. If called with the `R1` parameter, it will restore coolant output states to those saved during the most recent pause. This macro is called during the resume process to re-enable coolant.

Reviewer: ```suggestion
By default, this turns off any enabled coolant outputs. If called with the `R1` parameter, coolant output states will be restored to those saved during the most recent pause. This macro is called during the resume process to re-enable coolant.
```

; Usage
M3.9 [Snnnnn] [Pnn] [Dnnn]

; Start spindle assigned to the current tool at 8000 RPM
M3.9 S8000
```

Wrapping the inbuilt `M3` command, `M3.9` starts the spindle in the clockwise direction at a particular speed and waits for it to accelerate to the target speed.

If spindle feedback is configured, this command waits for a general-purpose input to change state before returning.

If spindle feedback is not configured, a static delay is used to make sure the spindle is up to speed before returning. The static speed used for this is timed by the operator during the wizard process, and modified based on the percentage speed change being requested. So if the RPM change is 50% of the maximum RPM of the spindle, then the delay will be 50% of the operator-recorded delay.

> Additionally, if Expert Mode is disabled, then this macro will pop up an operator confirmation/warning box when the spindle is going to accelerate from 0 rpm.

Reviewer: ```suggestion
Additionally, if Expert Mode is disabled, then this macro will pop up an operator confirmation when the spindle is going to accelerate from 0 rpm.
```
I'm not a fan of the no-space `/`, I think just having operator confirmation here is sufficient.
